docs/tutorial/object_decomp.md

```diff
@@ -1,14 +1,161 @@
-# Analysing object files
+# Object Decompilation
 
 Up: [Contents](contents.md)
+
 Previous: [Documenting](documenting.md)
 
-TODO: To be written once automated objects are present.
+Object decompilation is the process of taking an object file and writing instructions to extract the various assets from it in the correct format, with appropriate labelling to distinguish their nature and/or use.
+
+## What is an object file?
+
+An object file is generally where most of the information used by overlays is stored. Its contents can include
+
+- Vertices (positional/normal/color data used by displaylists)
+- Textures
+- DisplayLists (instructions to the graphics processor on how to put together textures and vertices)
+- Skeleton (The underlying structure of an actor's shape, that can be manipulated to change its "pose")
+- Animations
+- Texture Animations
+
+## How we work with objects
+
+Because these are regarded as copyrighted, we do not want them in the repository. We instead extract them from the ROM. To do this we use a system called ZAPD (Zelda Asset Processor for Decompilation). The main aim of object decompilation is to give ZAPD an XML file that tells it what the assets it is supposed to be extracting actually are, so it can put them in the right format and name them.
+
+## How to decomp an object
+
+Choose an object to decomp. As usual, some will be easier than others. For reasons explained shortly, it is much easier to decomp an object if all actors that use it are decompiled.
+
+### Files and folders
+
+Select the XML file of your selected object, which should be in `assets/xml/objects/object_name.xml`.
+
+The ZAPD output will go in the folder `assets/objects/object_name/`. You'll want this folder open later to check the output is correct.
+
+### Examining actor files
+
+All objects have had their XML generated in an automated way, so most constituent parts of each object are already identified, but will still need to be named and documented properly. Also, these objects usually have some blobs unreferenced by the object's own contents and hence not automatically extracted; most of the time these can be identified by looking at references in the actor which uses said object.
+
+### Extracting assets
+
+You can run `extract_assets.py` to extract the object's assets. Running it with no arguments extracts *everything* all over again, though. A better way is to run it with `-s` (for single file), and give it the location of the object you want to extract relative to `assets`, i.e.
```

Comment:
This is no longer true, it only extracts the files that have been updated (it tracks the modification times a bit like `make` does).